Documentation/contributing.rst

```diff
@@ -622,6 +622,76 @@ common unit test frameworks.
 A good use case for using ``AfterAll`` method is to remove containers or pods
 that are needed for multiple ``Its`` in the given ``Context`` or ``Describe``.
 
+JustAfterEach
+^^^^^^^^^^^^^
+
+This method will run just after each test and before ``AfterFailed`` and
+``AfterEach``. The main reason of this method is to provide a mechanism to do
+some assertions, for a group of test, after the test finished without
+duplicating some the code. For example, for each test a new a dead lock check
+runs to validate that nothing wrong happens during the execution of the test,
+that use to be a global ``JustAfterEach`` that validates after each test.
+
+AfterFailed
+^^^^^^^^^^^
+
+This method will run before all ``AfterEach`` and after ``JustAfterEach`` this
+function is only called when the test failed. The main reason of this code is to
+gather logs, cilium status, etc.. This helps to do some analysis when the system fails.
+
+Global execution
+^^^^^^^^^^^^^^^^
+
+Here is an example of the complete test case flow and when these methods will run:
+
+Test description diagram:
+::
+
+	Describe
+	BeforeAll(A)
```

Comment:
Indent things below `Describe` to show they are contained within the `Describe`.